docs/framework/react/guide/type-safety.md

```diff
@@ -115,6 +115,47 @@ const router = createRouter({
 
 As your application scales, TypeScript check times will naturally increase. There are a few things to keep in mind when your application scales to keep your TS check times down.
 
+### Only infer types you need
+
+A great pattern with client side data caches (TanStack Query etc.) is to prefetch data. For example with TanStack Query you might have a route which calls `ensureQueryData` in a `loader`.
```

Comment:
- Missing comma (`,`)  in "(TanStack Query, etc.) ".
- ... which calls `queryClient.ensureQueryData`